core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -1881,7 +1881,7 @@ class SparkContext(config: SparkConf) extends Logging {
           if (!fs.exists(hadoopPath)) {
             throw new FileNotFoundException(s"Jar ${path} not found")
           }
-          if (fs.isDirectory(hadoopPath)) {
+          if (fs.getFileStatus(hadoopPath).isDirectory) {
```

Comment:
@williamhyun, looks `fs.getFileStatus(hadoopPath).isDirectory` isn't exact synonym of `fs.isDirectory`, e.g.)

```java
  public boolean isDirectory(Path f) throws IOException {
    try {
      return getFileStatus(f).isDirectory();
    } catch (FileNotFoundException e) {
      return false;               // f does not exist
    }
  }
```

I think you can create a utility methods under `Utils.scala` to reuse.

Reply:
I missed that. Sorry. In that case, the utility methods will be identical with the deprecated Hadoop 3.2 `isFile/isDirectory` themselves. I'll close my PR. Thank you for your feedback.

Reply:
@williamhyun, I think they were deprecated to avoid calling redundant `getFileStatus` at [HADOOP-13321](https://issues.apache.org/jira/browse/HADOOP-13321):

> These methods have a habit of fostering inefficient call patterns in applications, resulting in multiple redundant getFileStatus calls.

 You can have one util that takes both `FileStatus` and `path` for example.

If you'd like to take more conservative, you can keep some valid calls (with `fs.exists()` as an example).